test/TestUtilities/Microsoft.Internal.NuGet.Testing.SignedPackages/CertificateUtilities.cs

```diff
@@ -47,5 +51,38 @@ public static X509Certificate2 GetCertificateWithPrivateKey(X509Certificate2 cer
             }
 #endif
         }
+
+        internal static byte[] GenerateSerialNumber(HashSet<BigInteger>? uniqueSerialNumbers = null)
+        {
+            // See https://www.rfc-editor.org/rfc/rfc5280#section-4.1.2.2
+            // A serial number MUST be:
+            //    * a non-negative integer
+            //    * unique for that CA
+            //    * <= 20 bytes
+            //    * big-endian encoded
+            byte[] bytes = new byte[20];
+
+            using (RandomNumberGenerator rng = RandomNumberGenerator.Create())
+            {
+                while (true)
+                {
+                    rng.GetBytes(bytes);
+
+                    // Treat the byte array as a little-endian integer.
+                    // Ensure the number is non-negative by setting the highest bit of the first byte to 0.
```

Comment:
Isn't this the last byte, evaluating to `bytes[19]` ?
```suggestion
                    // Ensure the number is non-negative by setting the highest bit of the last byte to 0.
```

Reply:
It's the first byte in little endian.  Updated the comment.